// @trq-fundacion/frontend-design-system/dist/cjs/index.js
"use strict";var e=require("react/jsx-runtime"),t=require("react"),r=require("styled-components"),i=require("@mui/material/Checkbox"),o=require("@mui/material/FormControlLabel"),n=require("@mui/material/Button"),a=require("@mui/icons-material/Warning"),s=require("@mui/material/Stack"),l=require("@mui/icons-material/Notifications"),c=require("@mui/material/Avatar"),d=require("@mui/material/Box"),m=require("@mui/material/Grid"),x=require("@mui/material/Link"),p=require("@mui/material/TextField"),u=require("@mui/material/Typography"),h=require("@mui/material/styles"),j=require("@mui/material/Card"),g=require("@mui/material/CardContent");function b(e){var t=Object.create(null);return e&&Object.keys(e).forEach((function(r){if("default"!==r){var i=Object.getOwnPropertyDescriptor(e,r);Object.defineProperty(t,r,i.get?i:{enumerable:!0,get:function(){return e[r]}})}})),t.default=e,Object.freeze(t)}var y=b(t);function f(e,t){var r={};for(var i in e)Object.prototype.hasOwnProperty.call(e,i)&&t.indexOf(i)<0&&(r[i]=e[i]);if(null!=e&&"function"==typeof Object.getOwnPropertySymbols){var o=0;for(i=Object.getOwnPropertySymbols(e);o<i.length;o++)t.indexOf(i[o])<0&&Object.prototype.propertyIsEnumerable.call(e,i[o])&&(r[i[o]]=e[i[o]])}return r}const v=r.div`
	background-color: white;
	border-radius: 5px;
	box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
	margin: 1rem;
	padding: 1rem;
	display: flex;
	flex-direction: column;
	width: 300px;
	transition: 0.3s;
	animation: ease-in-out;
	cursor: pointer;
	&:hover {
		transform: scale(1.1);
	}
`,O=r.h2`
	font-size: 1.25rem;
	color: black;
	margin-bottom: 0.5rem;
`,C=r.p`
	color: #757575;
	font-size: 0.875rem;
	margin-bottom: 0.5rem;
`,k=r.p`
	flex-grow: 1;
	font-size: 1rem;
	color: black;
	line-height: 1.5;
	text-align: justify;
`,w=r.button`
	background: #007bff;
	border: none;
	color: white;
	cursor: pointer;
	font-size: 0.875rem;
	margin-top: 0.5rem;
	padding: 0.5rem;
	border-radius: 5px;
	outline: none;
	opacity: 1;
	transition: opacity 0.3s ease-out;

	&:hover:enabled {
		opacity: 0.7;
	}

	&:disabled {
		background-color: #cccccc;
		color: #969696;
	}
`,B=r.div`
	display: flex;
	justify-content: flex-end;
`;r.div`
	width: 50%;
`;const q=t=>{var{children:r}=t,i=f(t,["children"]);return e.jsx(v,Object.assign({},i,{children:r}))};(q.Title=({title:t})=>e.jsx(O,{children:t})).displayName="CardBase.Title",(q.Author=({username:t})=>e.jsx(C,{children:t})).displayName="CardBase.Author",(q.Description=({description:t})=>e.jsx(k,{children:t})).displayName="CardBase.Description",(q.Button=t=>{var{text:r}=t,i=f(t,["text"]);return e.jsx(w,Object.assign({},i,{children:r}))}).displayName="CardBase.Button";const S=(e,t)=>e.length>t,N=(e,t)=>S(e,t)?e.substring(0,t)+"...":e,P=r=>{var{title:i,username:o,description:n}=r,a=f(r,["title","username","description"]);const[s,l]=t.useState(!1);return e.jsxs(q,Object.assign({},a,{children:[e.jsx(q.Title,{title:i}),e.jsx(q.Author,{username:o}),e.jsx(q.Description,{description:s?n:N(n,100)}),e.jsx(B,{children:e.jsx(q.Button,{text:s?"Show Less":"Show More",onClick:e=>{e.stopPropagation(),l(!s)},disabled:!S(n,100)})})]}))},A=r.div`
	display: flex;
	align-items: flex-start;
	gap: 8px;
	padding: 16px;
	border: 1px solid #ccc;
	border-radius: 4px;
	margin-bottom: 16px;
`,E=r.div`
	display: flex;
	flex-direction: column;
	gap: 4px;
`,F=r.h4`
	margin: 0;
	margin-bottom: 0.5rem;
	color: black;
`,T=r.p`
	margin: 0;
	font-size: 0.875rem;
	color: #757575;
	margin-bottom: 0.5rem;
`,z=r.p`
	margin: 0;
	color: black;
	text-align: justify;
`,I=r.img`
	width: 40px;
	height: 40px;
	border-radius: 50%;
	object-fit: cover;
	margin-right: 8px;
`,L=({children:t})=>e.jsx(A,{children:t});(L.Author=({name:t})=>e.jsx(F,{children:t})).displayName="Comment.Author",(L.Email=({email:t})=>e.jsx(T,{children:t})).displayName="Comment.Email",(L.Text=({text:t})=>e.jsx(z,{children:t})).displayName="Comment.Text",(L.Avatar=({src:t="https://cornerstoneia.com/wp-content/uploads/2019/08/avatar-placeholder.jpeg"})=>e.jsx(I,{src:t,alt:"Avatar"})).displayName="Comment.Avatar";const M=({author:t,text:r,email:i,avatarUrl:o})=>e.jsxs(L,{children:[e.jsx(L.Avatar,{src:o}),e.jsxs(E,{children:[e.jsx(L.Author,{name:t}),e.jsx(L.Email,{email:i}),e.jsx(L.Text,{text:r})]})]});function W(){return W=Object.assign?Object.assign.bind():function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var i in r)Object.prototype.hasOwnProperty.call(r,i)&&(e[i]=r[i])}return e},W.apply(this,arguments)}var D=function(e){return y.createElement("svg",W({xmlns:"http://www.w3.org/2000/svg",width:25,height:25},e),y.createElement("g",{transform:"translate(3.966 3.568)"},y.createElement("path",{d:"m0-.001 17.435 18.213M0 18.212 17.435 0",style:{opacity:1,fill:"#5f6368",stroke:"#5f6368",strokeWidth:3.23161912,strokeLinecap:"round",strokeMiterlimit:4,fillOpacity:1,strokeOpacity:1}})))};const H=r.keyframes`
  0% {
    box-shadow: 0px 0px 0px 0px rgb(213, 213, 213);
  }
  100% {
    box-shadow: 0px 0px 5px 25px rgba(35, 130, 220, 0);
  }
`,G=r.div`
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	background-color: rgba(0, 0, 0, 0.5);
	display: flex;
	justify-content: center;
	align-items: center;
	z-index: 1;
`,R=r.div`
	position: relative;
	width: 50%;
`,U=r.div`
	text-align: justify;
	margin-bottom: 2rem;
`,V=r.div`
	background-color: white;
	border-radius: 5px;
	padding: 20px;
	box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
`,$=r.h2`
	margin: 0;
	color: black;
	font-size: 1.5rem;
	font-weight: 600;
	margin-bottom: 1rem;
`,J=r.div`
	transform: scale(0.7);
	display: flex;
	position: absolute;
	top: 15px;
	right: 15px;
	cursor: pointer;
	padding: 5px;
	transition: background-color 0.3s ease-in-out;
	border-radius: 50%;
	&:hover {
		background-color: #dfdfdf;
	}
	&.pulse {
		animation: ${H} 0.5s ease-in-out;
	}
`,K=({children:r,isOpen:i=!1,onClose:o})=>{const[n,a]=(e=>{const[r,i]=t.useState(!1),[o,n]=t.useState(null);return t.useEffect((()=>()=>{o&&clearTimeout(o)}),[o]),[r,()=>{i(!0);const t=setTimeout((()=>{e(),i(!1)}),200);n(t)}]})(o);return e.jsx(e.Fragment,{children:i&&e.jsx(G,Object.assign({"data-testid":"modal-backdrop",onClick:o},{children:e.jsx(R,{children:e.jsxs(V,Object.assign({onClick:e=>e.stopPropagation()},{children:[e.jsx(J,Object.assign({className:n?"pulse":"",onClick:a,"data-testid":"modal-closeIcon"},{children:e.jsx(D,{})})),r]}))})}))})};(K.Header=({title:t})=>e.jsx($,{children:t})).displayName="Modal.Header",(K.Body=({children:t})=>e.jsx(U,{children:t})).displayName="Modal.Body",(K.Footer=({children:t})=>e.jsx("div",{children:t})).displayName="Modal.Footer";const Q=({title:t,body:r,footer:i,isOpen:o,onClose:n})=>e.jsxs(K,Object.assign({isOpen:o,onClose:n},{children:[e.jsx(K.Header,{title:t}),e.jsx(K.Body,{children:r}),e.jsx(K.Footer,{children:i})]})),X=t=>{var{type:r}=t,i=f(t,["type"]);return e.jsx(Q,Object.assign({},i))},Y=()=>e.jsx(o,{control:e.jsx(i,{value:"remember",color:"primary"}),label:"Recordar contraseña"}),Z=({text:t,color:r,onClick:i,isDisabled:o})=>e.jsx(n,Object.assign({variant:"contained",color:r,onClick:i,disabled:o},{children:t})),_=({comments:t})=>e.jsx(e.Fragment,{children:null==t?void 0:t.map((({id:t,name:r,email:i,body:o})=>e.jsx(M,{author:r,email:i,text:o},t)))}),ee=h.createTheme(),te=({title:t,description:r,onChangeEmail:i,onChangePassword:o})=>e.jsx(h.ThemeProvider,Object.assign({theme:ee},{children:e.jsx("div",{children:e.jsxs("div",{children:[e.jsxs("div",{children:[e.jsx(u,Object.assign({sx:{fontSize:14},color:"text.secondary",gutterBottom:!0},{children:t})),e.jsx(u,Object.assign({variant:"h5",component:"div"},{children:r}))]}),e.jsxs(d,Object.assign({component:"form",noValidate:!0,sx:{mt:1}},{children:[e.jsx(p,{margin:"normal",required:!0,fullWidth:!0,id:"email",label:"Correo electrónico",name:"email",autoComplete:"email",autoFocus:!0,onChange:i}),e.jsx(p,{margin:"normal",required:!0,fullWidth:!0,name:"password",label:"Contraseña",type:"password",id:"password",autoComplete:"current-password",onChange:o}),e.jsx(m,Object.assign({container:!0},{children:e.jsx(m,Object.assign({item:!0,xs:!0},{children:e.jsx(x,Object.assign({href:"#",variant:"body2"},{children:"¿Olvidaste la contraseña?"}))}))}))]}))]})})})),re=({children:t})=>e.jsx(j,{children:e.jsx(g,{children:t})});(re.Header=({children:t})=>e.jsx("div",{children:t})).displayName="SimpleCard.Header",(re.Body=({children:t})=>e.jsx("div",{children:t})).displayName="SimpleCard.Body",(re.Footer=({children:t})=>e.jsx("div",{children:t})).displayName="SimpleCard.Footer";const ie=({header:t,body:r,footer:i})=>e.jsxs(re,{children:[e.jsx(re.Header,{children:t}),e.jsx(re.Body,{children:r}),e.jsx(re.Footer,{children:i})]}),oe=t=>{var{type:r}=t,i=f(t,["type"]);return e.jsx(ie,Object.assign({},i))};exports.AvatarComponent=({src:t,alt:r})=>e.jsx(c,{src:t,alt:r}),exports.Card=t=>{var{type:r}=t,i=f(t,["type"]);return e.jsx(P,Object.assign({},i))},exports.CheckboxLogin=Y,exports.Comment=M,exports.Login=({secondaryButton:r,signInside:i,checkboxLogin:o,result:n,api:a})=>{const[s,l]=t.useState(""),[c,d]=t.useState(""),[m,x]=t.useState(!1);return e.jsx(oe,{header:e.jsx(e.Fragment,{}),body:e.jsxs("div",{children:[e.jsx(te,{title:i.title,description:i.description,onChangeEmail:e=>l(e.target.value),onChangePassword:e=>d(e.target.value)}),e.jsx(Y,{title:o.title})]}),footer:e.jsx(Z,{text:r.text,color:r.color,onClick:()=>"",isDisabled:m})})},exports.Modal=X,exports.Post=({title:t,body:r,comments:i,isOpen:o,onClose:n})=>e.jsx(X,{title:t,body:r,footer:e.jsx(_,{comments:i}),isOpen:o,onClose:n}),exports.PrimaryButton=({text:t,color:r,onClick:i})=>e.jsx(n,Object.assign({variant:"outlined",color:r,onClick:i},{children:t})),exports.SecondaryButton=Z,exports.WithStartIconEmergencyButton=({text:t,onClick:r})=>e.jsx(s,Object.assign({direction:"row",spacing:2},{children:e.jsx(n,Object.assign({variant:"contained",color:"error",onClick:r,startIcon:e.jsx(a,{})},{children:t}))})),exports.WithStartIconWarningButton=({text:t,onClick:r})=>e.jsx(s,Object.assign({direction:"row",spacing:2},{children:e.jsx(n,Object.assign({variant:"outlined",color:"warning",onClick:r,startIcon:e.jsx(l,{})},{children:t}))}));
